crates/holochain/src/core/workflow/sys_validation_workflow/validate_op_tests.rs

```diff
@@ -0,0 +1,1229 @@
+    holochain_trace::test_run().unwrap();
+
+    let mut test_case = TestCase::new().await;
+
+    // Previous action
+    let mut action = fixt!(Create);
+    action.author = test_case.agent.clone().into();
+    action.timestamp = Timestamp::now();
+    action.action_seq = 10;
+    action.prev_action = fixt!(ActionHash);
+    let previous_action = test_case.sign_action(Action::Create(action)).await;
+
+    // Op to validate
+    let mut create_action = fixt!(Create);
+    create_action.author = previous_action.action().author().clone();
+    create_action.action_seq = previous_action.action().action_seq() + 1;
+    create_action.prev_action = previous_action.as_hash().clone();
+    create_action.timestamp = Timestamp::now().into();
+    create_action.entry_type = EntryType::CapClaim;
+    let op = DhtOp::StoreRecord(
+        fixt!(Signature),
+        Action::Create(create_action),
+        holochain_zome_types::record::RecordEntry::NotStored,
+    );
+
+    let outcome = test_case
+        .expect_retrieve_actions_from_cascade(vec![previous_action])
+        .with_op(op)
+        .execute()
+        .await
+        .unwrap();
+
+    assert!(
+        matches!(outcome, Outcome::Accepted),
+        "Expected Accepted but actual outcome was {:?}",
+        outcome
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn validate_store_entry_with_entry_with_wrong_entry_type() {
+    holochain_trace::test_run().unwrap();
+
+    let mut test_case = TestCase::new().await;
+
+    // Previous action
+    let mut action = fixt!(Create);
+    action.author = test_case.agent.clone().into();
+    action.timestamp = Timestamp::now();
+    action.action_seq = 10;
+    action.prev_action = fixt!(ActionHash);
+    let previous_action = test_case.sign_action(Action::Create(action)).await;
+
+    // Op to validate
+    let app_entry = Entry::App(fixt!(AppEntryBytes));
+    let entry_hash = EntryHashed::from_content_sync(app_entry.clone());
+    let mut create_action = fixt!(Create);
+    create_action.author = previous_action.action().author().clone();
+    create_action.action_seq = previous_action.action().action_seq() + 1;
+    create_action.prev_action = previous_action.as_hash().clone();
+    create_action.timestamp = Timestamp::now().into();
+    create_action.entry_type = EntryType::AgentPubKey; // Claiming to be a public key but is actually an app entry
+    create_action.entry_hash = entry_hash.as_hash().clone();
+    let op = DhtOp::StoreRecord(
+        fixt!(Signature),
+        Action::Create(create_action),
+        holochain_zome_types::record::RecordEntry::Present(app_entry),
+    );
+
+    let outcome = test_case
+        .expect_retrieve_actions_from_cascade(vec![previous_action])
+        .with_op(op)
+        .execute()
+        .await
+        .unwrap();
+
+    assert!(
+        matches!(outcome, Outcome::Rejected),
+        "Expected Rejected but actual outcome was {:?}",
+        outcome
+    );
+}
+
+// TODO Flaky
```

Comment:
Oops, any idea why this is flaky?
I think we can ignore it while it's flaky.

Reply:
Yes, I was generating a mismatched entry randomly which of course doesn't have to be different. I'd done this in two other places so I've updated all of them to check the data is mismatched before using it